src/Calculator/BcMathCalculator.php

```diff
@@ -1,241 +1,212 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Money\Calculator;
 
+use InvalidArgumentException;
 use Money\Calculator;
 use Money\Money;
 use Money\Number;
 
-/**
- * @author Frederik Bosch <f.bosch@genkgo.nl>
- */
+use function bcadd;
+use function bccomp;
+use function bcdiv;
+use function bcmod;
+use function bcmul;
+use function bcsub;
+use function ltrim;
+
 final class BcMathCalculator implements Calculator
 {
-    /**
-     * @var string
-     */
-    private $scale;
+    private const SCALE = 14;
 
-    /**
-     * @param int $scale
-     */
-    public function __construct($scale = 14)
+    /** @psalm-pure */
+    public static function compare(string $a, string $b): int
     {
-        $this->scale = $scale;
+        return bccomp($a, $b, self::SCALE);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public static function supported()
+    /** @psalm-pure */
+    public static function add(string $amount, string $addend): string
     {
-        return extension_loaded('bcmath');
+        return bcadd($amount, $addend, self::SCALE);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function compare($a, $b)
+    /** @psalm-pure */
+    public static function subtract(string $amount, string $subtrahend): string
     {
-        return bccomp($a, $b, $this->scale);
+        return bcsub($amount, $subtrahend, self::SCALE);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function add($amount, $addend)
+    /** @psalm-pure */
+    public static function multiply(string $amount, string $multiplier): string
     {
-        return (string) Number::fromString(bcadd($amount, $addend, $this->scale));
+        return bcmul($amount, $multiplier, self::SCALE);
```

Comment:
I believe `add` and `substract` could return additional zeros in the fractional part, depending on the scale, and therefore the [Number object was used to get rid of them](https://github.com/moneyphp/money/blob/master/src/Number.php#L55). 

All operations should return a value without additional zeros, otherwise we would have to improve our equality comparison to allow for these zeros. My guess is that performance, apparently of high importance, would suffer more from additional operations during equality comparison than during calculator operations.

Reply:
It is possible to re-introduce `Number` here, at the expense of some performance improvements that would still be noticeable.

Reply:
Checked this again: indeed, equality is not as important as general-purpose multiplication/division, and `round()` operations already take care of this.

Marking therefore as resolved.